operator/cmd/configurator/bootstrap.go

```diff
@@ -0,0 +1,36 @@
+package configurator
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/clusterconfiguration"
+)
+
+// Template out the bootstrap file
+// This takes an input template, resolves any remaining external references, then writes out the resulting bootstrap file
+func templateBootstrapYaml(inFile, outFile string) error {
+	var template map[string]clusterconfiguration.ClusterConfigTemplateValue
+	buf, err := os.ReadFile(inFile)
+	if err != nil {
+		return fmt.Errorf("cannot load bootstrap template file: %w", err)
+	}
+	if err := json.Unmarshal(buf, &template); err != nil {
+		return fmt.Errorf("cannot parse bootstrap template file: %w", err)
+	}
+
+	var config []string
+	for k, v := range template {
+		// Work out what the value should be and add it to the output.
+		repr, err := clusterconfiguration.ExpandValueForTemplate(v)
+		if err != nil {
+			return fmt.Errorf("cannot resolve value %s: %w", k, err)
+		}
+		config = append(config, fmt.Sprintf("%s: %s", k, repr))
```

Comment:
This hurts me a bit to see but I think it'll be fine for all cases we're interested in. If anything, we could drop all of this into a `map` and implement `MarshalYAML/JSON` for the repr type so we could delegate to `yaml.Marshal`.

Reply:
I had a brief look at this - MarshalYAML returns an interface{} that's then marshalled out, which means if you want it to work then you need the type of the thing to return (which puts us back where we started). There didn't appear to be anything to hand that'd let me supply a []byte.

Reply:
(Yes, this made me wince too.)